backend/routes/discord.py

```diff
@@ -0,0 +1,83 @@
+"""Routes which directly interact with discord related data."""
+
+import pydantic
+from spectree import Response
+from starlette.authentication import requires
+from starlette.responses import JSONResponse
+from starlette.routing import Request
+
+from backend import discord, models, route
+from backend.validation import ErrorMessage, api
+
+NOT_FOUND_EXCEPTION = JSONResponse(
+    {"error": "Could not find the requested resource in the guild or cache."}, status_code=404
+)
+
+
+class RolesRoute(route.Route):
+    """Refreshes the roles database."""
+
+    name = "roles"
+    path = "/roles"
+
+    class RolesResponse(pydantic.BaseModel):
+        """A list of all roles on the configured server."""
+
+        roles: list[models.DiscordRole]
+
+    @requires(["authenticated", "admin"])
+    @api.validate(
+        resp=Response(HTTP_200=RolesResponse),
+        tags=["roles"]
+    )
+    async def patch(self, request: Request) -> JSONResponse:
+        """Refresh the roles database."""
+        roles = await discord.get_roles(request.state.db, force_refresh=True)
+
+        return JSONResponse(
+            {"roles": [role.dict() for role in roles]},
+        )
```

Comment:
Yeah here too - we could've just returned the Discord data directly without first turning it into a Pydantic model.

Reply:
I'm not sure what problem you're trying to solve here. There is no extra complexity to using the models if we already have them, as it's just some init/dict calls every now and then. The benefit to using them is we get built-in validation, and it's mentally easier to deal with than a bunch of random dicts all over the place.